From 9bd2dcc8a785d977cdf67a7cbb467423803e3327 Mon Sep 17 00:00:00 2001
From: David Edmundson <kde@davidedmundson.co.uk>
Date: Fri, 20 Sep 2024 14:36:11 +0200
Subject: [PATCH] startkde: Disable session management in our session
 management

We have two types of session restore, ksmserver for ICE clients and a
small fallback app that runs after ksmserver for any stragglers.

This fallback app is a gui application, as such it will try and connect
to ksmserver. This isn't useful and can cause a block.

BUG: 488853
---
 startkde/session-restore/restore.cpp | 1 +
 startkde/session-restore/save.cpp    | 1 +
 2 files changed, 2 insertions(+)

diff --git a/startkde/session-restore/restore.cpp b/startkde/session-restore/restore.cpp
index 2d86215a6f..aecc179139 100644
--- a/startkde/session-restore/restore.cpp
+++ b/startkde/session-restore/restore.cpp
@@ -22,6 +22,7 @@ using namespace Qt::StringLiterals;
 
 int main(int argc, char *argv[])
 {
+    QCoreApplication::setAttribute(Qt::AA_DisableSessionManager);
     QGuiApplication a(argc, argv);
     a.setDesktopSettingsAware(false);
     a.setApplicationName(u"plasmasessionrestore"_s);
diff --git a/startkde/session-restore/save.cpp b/startkde/session-restore/save.cpp
index a2a0fb2086..5e1473ed0b 100644
--- a/startkde/session-restore/save.cpp
+++ b/startkde/session-restore/save.cpp
@@ -29,6 +29,7 @@ using namespace Qt::StringLiterals;
 
 int main(int argc, char *argv[])
 {
+    QCoreApplication::setAttribute(Qt::AA_DisableSessionManager);
     QGuiApplication::setDesktopFileName(u"plasma-fallback-session-save"_s);
     QGuiApplication a(argc, argv);
     a.setApplicationName(u"plasmasessionrestore"_s);
-- 
GitLab

